src/org/zaproxy/zap/extension/formHandler/DialogAddToken.java

```diff
@@ -0,0 +1,233 @@
+                .addComponent(enabledLabel)
+                .addComponent(getEnabledCheckBox()))
+        );
+
+        return fieldsPanel;
+    }
+
+    @Override
+    protected String getConfirmButtonLabel() {
+        return CONFIRM_BUTTON_LABEL;
+    }
+
+    @Override
+    protected void init() {
+        getNameTextField().setText("");
+        getValueTextField().setText("");
+        getEnabledCheckBox().setSelected(true);
+        token = null;
+    }
+
+    /*
+     * Confirms the fields do not already exist
+     */
+    @Override
+    protected boolean validateFields() {
+        String tokenName = getNameTextField().getText().toLowerCase();
+        for (FormHandlerParamToken t : tokens) {
+            if (tokenName.equals(t.getName())) {
+                JOptionPane.showMessageDialog(this, TEXT_NAME_REPEATED_DIALOG,
+                        TITLE_NAME_REPEATED_DIALOG,
+                        JOptionPane.INFORMATION_MESSAGE);
+                getNameTextField().requestFocusInWindow();
+                return false;
+            }
+        }
+
+        return true;
+    }
+
+    /**
+     * When the Add button is clicked, create a new token.
+     * This token will be created with the name, value and enabled input by the user.
+     * The name will always be lower case.
+     */
+    @Override
+    protected void performAction() {
+        token = new FormHandlerParamToken(getNameTextField().getText().toLowerCase(), getValueTextField().getText(), getEnabledCheckBox().isSelected());
+    }
+
+    @Override
+    protected void clearFields() {
+        getNameTextField().setText("");
+        getNameTextField().discardAllEdits();
+    }
+
+    public FormHandlerParamToken getToken() {
+        return token;
+    }
+
+    protected ZapTextField getNameTextField() {
+        if (nameTextField == null) {
+            nameTextField = new ZapTextField(25);
+            nameTextField.getDocument().addDocumentListener(new DocumentListener() {
+
+                @Override
+                public void removeUpdate(DocumentEvent e) {
+                    checkAndEnableConfirmButton();
+                }
+
+                @Override
+                public void insertUpdate(DocumentEvent e) {
+                    checkAndEnableConfirmButton();
+                }
+
+                @Override
+                public void changedUpdate(DocumentEvent e) {
+                    checkAndEnableConfirmButton();
+                }
+
+                private void checkAndEnableConfirmButton() {
+                    setConfirmButtonEnabled(getNameTextField().getDocument().getLength() > 0);
+                }
+            });
+        }
+
+        return nameTextField;
+    }
+
+    protected ZapTextField getValueTextField() {
+        if (valueTextField == null) {
+            valueTextField = new ZapTextField(25);
+            valueTextField.getDocument().addDocumentListener(new DocumentListener() {
```

Comment:
This document listener could be removed, it's not needed, the name field already has its own listener.